pkg/core/conn.go

```diff
@@ -96,21 +96,26 @@ func (c *Conn) Send(msgType wasabi.MessageType, msg []byte) error {
 		return c.clientConn.Send(msgType, msg)
 	}
 
+	if !c.DoneRequest(respID.ReqID, msg) {
+		return c.clientConn.Send(msgType, msg)
+	}
+
+	return nil
+}
+
+func (c *Conn) DoneRequest(reqID int64, resp []byte) bool {
 	c.mu.Lock()
-	ch, ok := c.requests[respID.ReqID]
-	delete(c.requests, respID.ReqID)
+	ch, ok := c.requests[reqID]
+	delete(c.requests, reqID)
```

Comment:
should we delete only if `ok == true`?

Reply:
@shan-96 yeah you're right, I didn't like how code was looking  what we need to branch before release lock and do unlocking in both code branches...  let me fix this

Reply:
@ksysoev hang on, just checked `delete`'s documentation. It seems it never throws any error and is able to handle both cases when map is nil or there is no such element present. I guess you can keep it this way